[TensorExpr] Run constant pooling in fusion groups to dedupe constants.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#47402 [TensorExpr] Run constant pooling in fusion groups to dedupe constants.**

Differential Revision: [D24740957](https://our.internmc.facebook.com/intern/diff/D24740957)

Commit:
[TensorExpr] Run constant pooling in fusion groups to dedupe constants.

[ghstack-poisoned]

eellison reviewed (approved):
nit: this might be better run we inline small graphs, since it would avoid instantiating an AliasDb. We don't count constants in `blockSize` so deferring  constant  pooling shouldn't affect whether we inline.

Commit:
Update on "[TensorExpr] Run constant pooling in fusion groups to dedupe constants."


Differential Revision: [D24740957](https://our.internmc.facebook.com/intern/diff/D24740957)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Run constant pooling in fusion groups to dedupe constants."


Differential Revision: [D24740957](https://our.internmc.facebook.com/intern/diff/D24740957)

[ghstack-poisoned]
